tests/state_test_util.go

```diff
@@ -253,9 +256,19 @@ func (t *StateTest) RunNoVerify(tx kv.RwTx, subtest StateSubtest, vmconfig vm.Co
 		context.BaseFee = new(uint256.Int)
 		context.BaseFee.SetFromBig(baseFee)
 	}
-	if t.json.Env.Random != nil {
+	if t.json.Env.Difficulty != nil {
+		context.Difficulty = new(big.Int).Set(t.json.Env.Difficulty)
+	}
+	if config.IsLondon(0) && t.json.Env.Random != nil {
 		rnd := libcommon.BigToHash(t.json.Env.Random)
 		context.PrevRanDao = &rnd
+		context.Difficulty = big.NewInt(0)
+	}
+	if config.IsCancun(0) && t.json.Env.ExcessBlobGas != nil {
```

Comment:
Should be `config.IsCancun(block.Time())` instead of `config.IsCancun(0)` because, starting with Shanghai, forks are triggered by the time rather than the number of the block.